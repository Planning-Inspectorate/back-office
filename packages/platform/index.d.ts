import './config/dotenv';
import './http/ttl';
import './util/CryptoUtils';
import './util/date';
import './util/filter';
import './util/url';
import './validators/date';
import './validators/postcode';

declare module '@pins/platform' {
	export * from './config/dotenv';
	export * from './http/ttl';
	export * from './util/CryptoUtils';
	export * from './util/date';
	export * from './util/filter';
	export * from './util/url';
	export * from './validators/date';
	export * from './validators/postcode';
}
