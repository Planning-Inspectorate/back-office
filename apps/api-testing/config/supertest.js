// @ts-nocheck
import supertest from 'supertest';
import config from './config.cjs';
import { Logger } from '../utils/logger.js';

const logger = new Logger();
logger.log('green', '************* Tests starting ****************');
logger.log('cyan', `Running tests against: ${config.baseUrl}`);
logger.log('green', '*********************************************');

const request = supertest(config.baseUrl);

export default request;
