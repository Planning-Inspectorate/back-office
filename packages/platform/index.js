export * from './http/hooks.js';
export * from './testing/assets/assets.js';
export * from './testing/fake.js';
export * from './testing/html-parser.js';
export * from './utils/date.js';
export * from './utils/dotenv.js';
export * from './utils/filter.js';
export * from './validators/date.js';
export * from './validators/postcode.js';
