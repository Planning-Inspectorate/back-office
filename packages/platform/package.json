{
  "name": "@pins/platform",
  "version": "0.0.0",
  "type": "module",
  "main": "index.js",
  "private": true,
  "engines": {
    "node": ">=16.14.0"
  },
  "scripts": {
    "clean": "rimraf .turbo node_modules dist",
    "lint:js": "npx eslint .",
    "lint:js:fix": "npx eslint . --fix",
    "tscheck": "npx tsc -p jsconfig.json --maxNodeModuleJsDepth 0"
  },
  "dependencies": {
    "date-fns": "2.28.0",
    "dotenv": "16.0.0",
    "got": "^12.5.2",
    "html": "^1.0.0",
    "lodash-es": "^4.17.21",
    "node-html-parser": "^5.3.3",
    "randexp": "^0.5.3"
  },
  "devDependencies": {
    "@types/html": "^1.0.1",
    "@types/lodash-es": "^4.17.6",
    "type-fest": "^2.12.2"
  }
}
