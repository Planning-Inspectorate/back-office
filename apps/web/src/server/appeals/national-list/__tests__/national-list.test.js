import { parseHtml } from '@pins/platform';
import nock from 'nock';
import supertest from 'supertest';
import { createTestEnvironment } from '../../../../../testing/index.js';

const { app, installMockApi, teardown } = createTestEnvironment();
const request = supertest(app);
const baseUrl = '/appeals-service/appeals-list';

describe('national-list', () => {
	beforeEach(installMockApi);
	afterEach(teardown);

	describe('GET /', () => {
		it('should render national list', async () => {
			nock('http://test/').get('/appeals/').reply(200, []);

			const response = await request.get(baseUrl);
			const element = parseHtml(response.text);

			expect(element.innerHTML).toMatchSnapshot();
		});
	});
});
