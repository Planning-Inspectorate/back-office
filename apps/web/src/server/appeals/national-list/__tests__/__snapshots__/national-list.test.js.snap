// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`national-list GET / should render national list 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\">
    <div class=\\"govuk-grid-row\\">
        <div class=\\"govuk-grid-column-full\\">
            <h1 class=\\"govuk-heading-xl\\">Appeals</h1>
        </div>
    </div><span class=\\"govuk-caption-xl\\">Case officer</span>
    <h2 class=\\"govuk-heading-l govuk-!-margin-bottom-3\\">All cases</h2>
    <table class=\\"govuk-table\\">
        <thead class=\\"govuk-table__head\\">
            <tr class=\\"govuk-table__row\\">
                <th scope=\\"col\\" class=\\"govuk-table__header\\">Appeal ID</th>
                <th scope=\\"col\\" class=\\"govuk-table__header\\">Site address</th>
                <th scope=\\"col\\" class=\\"govuk-table__header\\">Appeal type</th>
                <th scope=\\"col\\" class=\\"govuk-table__header\\">Local planning authority</th>
                <th scope=\\"col\\" class=\\"govuk-table__header\\">Status</th>
            </tr>
        </thead>
        <tbody class=\\"govuk-table__body\\">
            <tr class=\\"govuk-table__row\\">
                <td class=\\"govuk-table__cell\\"><a class=\\"govuk-link\\" href=\\"/appeals-service/appeal-details/1\\"> 943245</a>
                </td>
                <td class=\\"govuk-table__cell\\">Copthalls, Clevedon Road, West Hill, BS48 1PN</td>
                <td class=\\"govuk-table__cell\\">Householder Appeal</td>
                <td class=\\"govuk-table__cell\\">Bradford Council</td>
                <td class=\\"govuk-table__cell\\"><strong class=\\"govuk-tag govuk-tag--\\">received_appeal</strong>
                </td>
            </tr>
            <tr class=\\"govuk-table__row\\">
                <td class=\\"govuk-table__cell\\"><a class=\\"govuk-link\\" href=\\"/appeals-service/appeal-details/2\\"> 129285</a>
                </td>
                <td class=\\"govuk-table__cell\\">19 Beauchamp Road, Bristol, BS7 8LQ</td>
                <td class=\\"govuk-table__cell\\">Householder Appeal</td>
                <td class=\\"govuk-table__cell\\">Bradford Council</td>
                <td class=\\"govuk-table__cell\\"><strong class=\\"govuk-tag govuk-tag--\\">received_appeal</strong>
                </td>
            </tr>
        </tbody>
    </table>
</main>"
`;
